### ISSUE TITLE
Repositories level environment variables

### ISSUE THREAD

**Issue by l3onL** [woodpecker-ci/woodpecker#2604]
### Clear and concise description of the problem

Use docker:cli with compose file to start host's containers.
When use docker compose deploy, often the image tag name come with ${CI_COMMIT_SHA:0:5}. When pipeline starts, use when to decide if should build images. If the image not build, but the variable still comes in. So the docker compose fails.

### Suggested solution

Add repositories level environment variables ${A} which is updatable to replace the ${CI_COMMIT_SHA:0:5}, then the pipeline meets the when condition will update the variable ${A} with ${CI_COMMIT_SHA:0:5}. In the future pipeline starts, even not build images, but the ${A} will point to a valid ${CI_COMMIT_SHA:0:5}.

Thanks

### Validations

- [X] Checked that the feature isn't part of the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't already an [issue](https://github.com/woodpecker-ci/woodpecker/issues) that request the same feature to avoid creating a duplicate.

**Comment by pat-s** [comment]
You can set environment variables on the pipeline level and use the `evaluate` when trigger to condition runs: https://woodpecker-ci.org/docs/next/usage/workflow-syntax#evaluate

**Comment by l3onL** [comment]
Set in the pipeline level does not help in the situation describe above. Set the Repositories level, will carry latest pipeline info. It may open a new potential flexibility.

**Comment by pat-s** [comment]
Sorry I don't understand what you mean. Can you rephrase the problem in general in better English?

**Comment by l3onL** [comment]
Sorry for my bad english. Thanking woodpecker yaml workflow as a programing language. There are two scope, global and pipeline level. By add repositories level, make the workflow has 3 scopes. Also I don't believe you can dynamically create global variables and update it (correct me here if I am wrong). You can only update it in the steps, that means these updated global variables won't persist.  By having repositories level and updatable variables will not only not contaminate the global scope, but also make it more flexible, just like programing language. In my case I just wan't keep last build ${CI_COMMIT_SHA:0:5} image which meet certain criteria in a variable, so that in later pipeline docker compose know which image it can pull.
